### ISSUE TITLE
Some users are healty and some are stuck on unhealty status(Degraded)

### ISSUE THREAD

**Issue by saurabh21316** [elastic/elastic-agent#3846]
![Screenshot 2023-11-29 at 3 37 03 PM](https://github.com/elastic/elastic-agent/assets/51045014/e1bca391-375a-4a3b-ad01-fadebb343bcf)
Hello Team,

I would really appreciate If I can get help to resolved this issue. I see few macs are getting health status and few are in unhealthy status.

First, I don't see logs under Logs Tab under Agent Details, so this is definitely bug
Second, I see few  users are healthy and some are stuck on unhealthy status. I don't where to start looking at the logs on the fleet server on elastic search. 

I have attached screenshot here. I am using version 8.11.1 ES and 8.11.0 Elastic Endpoint. Tried to uninstall and reinstall it but didn't help.  

PS: no firewalll policy blocking it.

Thanks!

**Comment by saurabh21316** [comment]
![Screenshot 2023-11-29 at 3 44 32 PM](https://github.com/elastic/elastic-agent/assets/51045014/2c0aff80-50e2-42a8-8d07-e89856518b4c)

**Comment by saurabh21316** [comment]
fleet
│ └─ status: (HEALTHY) Connected
└─ elastic-agent
  ├─ status: (DEGRADED) 1 or more components/units in a degraded state
  └─ endpoint-default
   ├─ status: (HEALTHY) Healthy: communicating with endpoint service
   ├─ endpoint-default
   │ └─ status: (DEGRADED) Applied policy {42b254d8-60c5-49fa-8ae5-8374d7ba4f81}
   └─ endpoint-default-42b254d8-60c5-49fa-8ae5-8374d7ba4f81
     └─ status: (DEGRADED) Applied policy {42b254d8-60c5-49fa-8ae5-8374d7ba4f81}

**Comment by pierrehilbert** [comment]
Hello,
Did you check logs locally on the Agent to try to get more information about what is happening here?

**Comment by saurabh21316** [comment]
@pierrehilbert, thank you for your prompt response. I've investigated the issue based on your suggestions. I've reviewed the logs and identified the root cause. Currently, my FLEET output is directed to the ES cluster on port 9200 (e.g., https://elastic-cluster.abc.com:9200). However, the challenge lies in the fact that my Elasticsearch cluster is located in a private subnet, not accessible from the internet. Consequently, the Elastic agent is unable to establish communication with the Elasticsearch cluster.

Here's my existing configuration:

Elasticsearch Cluster (Private Subnet in GCP) --> Fleet Server (private/public IP, exposed to the internet on port 8220, behind the firewall) --> Elastic Agent (on the internet)

I've also observed that I cannot set Logstash as the default output setting, which seems to limit my options. Thus, I find myself at an impasse. I'm seeking advice on how to resolve this issue without exposing the Elasticsearch cluster to the internet.

Your insights on this matter would be greatly appreciated.

Thanks!!
